meteor/client/ui/Settings/Studio/Routings.tsx

```diff
@@ -12,723 +11,903 @@ import {
 }
-interface IStudioRoutingsState {
-	editedItems: Array<string>
-}
-
-export const StudioRoutings = withTranslation()(
-	class StudioRoutings extends React.Component<Translated<IStudioRoutingsProps>, IStudioRoutingsState> {
-		constructor(props: Translated<IStudioRoutingsProps>) {
-			super(props)
 
-			this.state = {
-				editedItems: [],
-			}
-		}
-		isItemEdited = (routeSetId: string) => {
-			return this.state.editedItems.indexOf(routeSetId) >= 0
-		}
-		finishEditItem = (routeSetId: string) => {
-			const index = this.state.editedItems.indexOf(routeSetId)
-			if (index >= 0) {
-				this.state.editedItems.splice(index, 1)
-				this.setState({
-					editedItems: this.state.editedItems,
-				})
-			}
-		}
-		editItem = (routeSetId: string) => {
-			if (this.state.editedItems.indexOf(routeSetId) < 0) {
-				this.state.editedItems.push(routeSetId)
-				this.setState({
-					editedItems: this.state.editedItems,
-				})
-			} else {
-				this.finishEditItem(routeSetId)
-			}
-		}
-		confirmRemoveEGroup = (eGroupId: string, exclusivityGroup: StudioRouteSetExclusivityGroup) => {
-			const { t } = this.props
-			doModalDialog({
-				title: t('Remove this Exclusivity Group?'),
-				yes: t('Remove'),
-				no: t('Cancel'),
-				onAccept: () => {
-					this.removeExclusivityGroup(eGroupId)
-				},
-				message: (
-					<React.Fragment>
-						<p>
-							{t(
-								'Are you sure you want to remove exclusivity group "{{eGroupName}}"?\nRoute Sets assigned to this group will be reset to no group.',
-								{
-									eGroupName: exclusivityGroup.name,
-								}
-							)}
-						</p>
-						<p>{t('Please note: This action is irreversible!')}</p>
-					</React.Fragment>
-				),
-			})
-		}
-		confirmRemoveRoute = (routeSetId: string, route: RouteMapping, index: number) => {
-			const { t } = this.props
-			doModalDialog({
-				title: t('Remove this Route from this Route Set?'),
-				yes: t('Remove'),
-				no: t('Cancel'),
-				onAccept: () => {
-					this.removeRouteSetRoute(routeSetId, index)
-				},
-				message: (
-					<React.Fragment>
-						<p>
-							{t('Are you sure you want to remove the Route from "{{sourceLayerId}}" to "{{newLayerId}}"?', {
-								sourceLayerId: route.mappedLayer,
-								newLayerId: route.outputMappedLayer,
-							})}
-						</p>
-						<p>{t('Please note: This action is irreversible!')}</p>
-					</React.Fragment>
-				),
-			})
-		}
-		confirmRemove = (routeSetId: string) => {
-			const { t } = this.props
-			doModalDialog({
-				title: t('Remove this Route Set?'),
-				yes: t('Remove'),
-				no: t('Cancel'),
-				onAccept: () => {
-					this.removeRouteSet(routeSetId)
+export function StudioRoutings({
+	translationNamespaces,
+	studio,
+	studioMappings,
+	manifest,
+}: Readonly<IStudioRoutingsProps>): React.JSX.Element {
+	const { t } = useTranslation()
+	const { toggleExpanded, isExpanded } = useToggleExpandHelper()
+
+	const getRouteSetsFromOverrides = React.useMemo(
```

Comment:
I'm not a fan of the name of this. It starting with `get` suggests to me that it is a function, rather than the object/array that it is

Reply:
Complete agree

Reply:
fixed